### PR TITLE
Update npm dependencies and harden Dockerfile

### PR DESCRIPTION
**Pull request by fmarier**
The only "real" change here is the new version of extension-whitelist (to pull in https://github.com/brave/extension-whitelist/pull/59).

The rest are cleanups and should :crossed_fingers: not affect anything.

### CODE REVIEW

**Comment by petemill** [comment]
Hmm, CI failed...

**Comment by fmarier** [comment]
> Hmm, CI failed...

Yeah, I'm still trying to figure out why.

**Comment by fmarier** [comment]
This is now ready to be reviewed. Apologies for the noise.

**Review by onyb — APPROVED** [review]
++ for ethereum-remote-client and brave-wallet-lists update.

File: package.json [hunk 1]
```diff
@@ -85,7 +85,7 @@
     "url": "https://github.com/brave/brave-core-crx-packager.git"
   },
   "resolutions": {
-    "ansi-regex": "6.0.1"
+    "ansi-regex": "5.0.1"
```
Comment (diracdeltas) [inline package.json:88]: is this downgrade intentional?

Comment (fmarier) [reply package.json:88]: Yes, that was the cause of the build failures in https://github.com/brave/brave-core-crx-packager/pull/412.

From what I could tell, forcing this new version (done by Renovate in https://github.com/brave/brave-core-crx-packager/pull/325) is causing incompatibilities. It's also not required to fix any `npm audit` failures anymore.

**Review by diracdeltas — APPROVED** [review]

**Pull request merged**